docs/reference/builder.md

```diff
@@ -1396,6 +1396,31 @@ To use these, simply pass them on the command line using the flag:
 --build-arg <varname>=<value>
 ```
 
+These commands are exluded from the output of `docker history` by default.
+This reduces the risk of accidentally leaking sensitive authentication information
+in an `HTTP_PROXY` variable. This behaviour can be overriden by providing an
+explicit `ARG` instruction in the Dockerfile.
+
+For example, consider the following Dockerfiles that were both built using
+`--build-arg HTTP_PROXY=http://user:pass@proxy.example.com`
+
+```
+1 FROM ubuntu
+2 RUN echo "Hello World"
+```
+
+In this case, the value of the `HTTP_PROXY` variable is not available in the
+`docker history`.
+
+
+```
+1 FROM ubuntu
```

Comment:
Since we're not refering to specific lines from the text, can you;

- remove the line numbers
- add a `Dockerfile` syntax-hint;

        ```Dockerfile
        FROM ubuntu
        ARG HTTP_PROXY